packages/sdk-v2/src/sdk/savings/react.ts

```diff
@@ -63,9 +67,9 @@ export function useSavingsBalance(refresh: QueryParams["refresh"] = "never", env
   return { g$Balance: balance, savingsBalance: sBalance };
 }
 
-export const useSavingsFunctions = () => {
-  const gooddollar = useGetContract("GoodDollar", false, "savings") as IGoodDollar;
-  const gdStaking = useGetContract("GoodDollarStaking", false, "savings") as GoodDollarStaking;
+export const useSavingsFunctions = (requestedChainId: number) => {
```

Comment:
"writing" should not require a requestedchainId, writing is always done using the connected user wallet

Reply:
it still should, else it will try to useGetContract(GoodDollarStaking) on any connected network, which would fail on mainnet

Reply:
is removed again